src/table/use-sticky-columns-context.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { createContext, useContext, useLayoutEffect, useCallback, useRef, ReactNode } from 'react';
+import { unstable_batchedUpdates } from 'react-dom';
+
+interface StickyColumnsContextValue {
+  subscribe: (callback: (entry: any) => void) => void;
+  unsubscribe: (callback: (entry: any) => void) => void;
+}
+
+const StickyColumnsContext = createContext<StickyColumnsContextValue | null>(null);
+
+export const useStickyColumnsContext = () => {
+  const context = useContext(StickyColumnsContext);
+  if (!context) {
+    return { subscribe: () => {}, unsubscribe: () => {} };
+  }
+  const { subscribe, unsubscribe } = context as StickyColumnsContextValue;
+  return { subscribe, unsubscribe };
+};
+
+interface StickyColumnsContextProviderProps {
+  children: ReactNode;
+  wrapperRef: React.RefObject<HTMLDivElement>;
+  tableRef: React.RefObject<HTMLTableElement>;
+  shouldDisableStickyColumns: boolean;
+}
+interface StickyColumnsContextProvider {
+  (props: StickyColumnsContextProviderProps): JSX.Element;
+}
+
+export const StickyColumnsContextProvider: StickyColumnsContextProvider = ({
+  children,
+  wrapperRef,
+  tableRef,
+  shouldDisableStickyColumns,
+}) => {
+  const callbacks = useRef<Set<(entry: any) => void>>(new Set());
+
+  const checkStickyState = React.useCallback(() => {
```

Comment:
Minor (style): it would be nice to follow one particular styles for React imports like `useRef, useCallback` or `React.useRef, React.useCallback`